pallets/oracle/src/dia.rs

```diff
@@ -0,0 +1,207 @@
+					primitives::TokenSymbol::AMPE => unimplemented!(),
+				},
+				CurrencyId::ForeignAsset(_) => unimplemented!(),
+				CurrencyId::Native => unimplemented!(),
+				CurrencyId::StellarNative => unimplemented!(),
+				CurrencyId::AlphaNum4 { .. } => unimplemented!(),
+				CurrencyId::AlphaNum12 { .. } => unimplemented!(),
+			},
+			OracleKey::FeeEstimation => Some((vec![6u8], vec![])),
+		}
+	}
+}
+
+impl Convert<(Vec<u8>, Vec<u8>), Option<OracleKey>> for DiaOracleKeyConvertor {
+	fn convert(dia_oracle_key: (Vec<u8>, Vec<u8>)) -> Option<OracleKey> {
+		let (blockchain, symbol) = dia_oracle_key;
+		let blockchain = String::from_utf8(blockchain);
+		let symbol = String::from_utf8(symbol);
+		match (blockchain, symbol) {
+			(Ok(blockchain), Ok(symbol)) => {
+				if blockchain == DOT_DIA_BLOCKCHAIN && symbol == DOT_DIA_SYMBOL {
+					return Some(OracleKey::ExchangeRate(CurrencyId::Token(
+						primitives::TokenSymbol::DOT,
+					)))
+				} else if blockchain == KSM_DIA_BLOCKCHAIN && symbol == KSM_DIA_SYMBOL {
+					return Some(OracleKey::ExchangeRate(CurrencyId::Token(
+						primitives::TokenSymbol::KSM,
+					)))
+				} else {
+					return None
+				}
+			},
+			(_, _) => return None,
+		}
+	}
+}
+
+pub struct DiaOracleAdapter<
+	DiaPallet: DiaOracle,
+	UnsignedFixedPoint,
+	Moment,
+	ConvertKey,
+	ConvertPrice,
+	ConvertMoment,
+>(
+	marker::PhantomData<(
+		DiaPallet,
+		UnsignedFixedPoint,
+		Moment,
+		ConvertKey,
+		ConvertPrice,
+		ConvertMoment,
+	)>,
+);
+
+impl<Dia, UnsignedFixedPoint, Moment, ConvertKey, ConvertPrice, ConvertMoment>
+	DataProviderExtended<OracleKey, TimestampedValue<UnsignedFixedPoint, Moment>>
+	for DiaOracleAdapter<Dia, UnsignedFixedPoint, Moment, ConvertKey, ConvertPrice, ConvertMoment>
+where
+	Dia: DiaOracle,
+	ConvertKey: Convert<OracleKey, Option<(Vec<u8>, Vec<u8>)>>
+		+ Convert<(Vec<u8>, Vec<u8>), Option<OracleKey>>,
+	ConvertPrice: Convert<u128, Option<UnsignedFixedPoint>>,
+	ConvertMoment: Convert<u64, Option<Moment>>,
+{
+	fn get_no_op(key: &OracleKey) -> Option<TimestampedValue<UnsignedFixedPoint, Moment>> {
+		let dia_key: Option<(Vec<u8>, Vec<u8>)> = ConvertKey::convert(key.clone());
+		let Some((blockchain,symbol)) = dia_key else {
+            return None;
+        };
+
+		let Ok(coin_info) = Dia::get_coin_info(blockchain, symbol) else {
+            return None;
+        };
+
+		let Some(value) = ConvertPrice::convert(coin_info.price) else{
+            return None;
+        };
+		let Some(timestamp) = ConvertMoment::convert(coin_info.last_update_timestamp) else{
+            return None;
+        };
+
+		Some(TimestampedValue { value, timestamp })
+	}
+
+	///do not need this funtion implementation
+	fn get_all_values() -> Vec<(OracleKey, Option<TimestampedValue<UnsignedFixedPoint, Moment>>)> {
+		panic!("spacewalk oracle extension does not requre implementation of DataProviderExtended get_all_values function")
+	}
+}
+
+pub struct MockDiaOracleConvertor;
```

Comment:
This should maybe be called `MockOracleKeyConvertor` since that's what it's doing.

Reply:
Done.